debug_utils/sierra-emu/src/vm/circuit.rs

```diff
@@ -193,65 +256,86 @@ pub fn eval_get_output(
     let gate_offset = *circuit_info.values.get(&_info.output_ty).unwrap();
     let output_idx = gate_offset - 1 - circuit_info.n_inputs;
     let output = outputs[output_idx].to_owned();
-    let output_big = output.to_bigint().unwrap();
-
-    let mask: BigInt = BigInt::from_bytes_be(Sign::Plus, &[255; 12]);
-
-    let l0: BigInt = &output_big & &mask;
-    let l1: BigInt = (&output_big >> 96) & &mask;
-    let l2: BigInt = (&output_big >> 192) & &mask;
-    let l3: BigInt = (output_big >> 288) & &mask;
 
-    let range = BigInt::ZERO..(BigInt::from(1) << 96);
-    let vec_values = vec![
-        Value::BoundedInt {
-            range: range.clone(),
-            value: l0,
-        },
-        Value::BoundedInt {
-            range: range.clone(),
-            value: l1,
-        },
-        Value::BoundedInt {
-            range: range.clone(),
-            value: l2,
-        },
-        Value::BoundedInt { range, value: l3 },
-    ];
+    let output_struct = u384_to_struct(output);
+    let modulus = u384_to_struct(modulus);
 
-    EvalAction::NormalBranch(0, smallvec![Value::Struct(vec_values), Value::Unit])
+    EvalAction::NormalBranch(
+        0,
+        smallvec![
+            output_struct.clone(),
+            Value::Struct(vec![output_struct, modulus]),
+        ],
+    )
 }
 
-pub fn eval_u96_limbs_less_than_guarantee_verify(
+fn eval_u96_limbs_less_than_guarantee_verify(
     _registry: &ProgramRegistry<CoreType, CoreLibfunc>,
-    _info: &ConcreteU96LimbsLessThanGuaranteeVerifyLibfunc,
-    _args: Vec<Value>,
+    info: &ConcreteU96LimbsLessThanGuaranteeVerifyLibfunc,
+    args: Vec<Value>,
 ) -> EvalAction {
-    EvalAction::NormalBranch(0, smallvec![Value::Unit])
+    let [Value::Struct(garantee)]: [Value; 1] = args.try_into().unwrap() else {
+        panic!()
+    };
+    let limb_count = info.limb_count;
+    let Value::Struct(gate) = garantee.first().unwrap() else {
+        panic!();
+    };
+    let Value::Struct(modulus) = garantee.get(1).unwrap() else {
+        panic!();
+    };
```

Comment:
I think this is a better easier to read:
```suggestion
    let Value::Struct(gate) = garantee[0] else {
        panic!();
    };
    let Value::Struct(modulus) = garantee[1].unwrap() else {
        panic!();
    };
```